bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -2343,18 +2345,22 @@ void ReplicaImp::onMessage<CheckpointMsg>(CheckpointMsg *msg) {
       // <= to allow repeating checkpoint message since state transfer may not kick in when we are inside active
       // window
       if (pos != tableOfStableCheckpoints.end()) delete pos->second;
-      CheckpointMsg *x = new CheckpointMsg(msgGenReplicaId, msgSeqNum, msgState, stateDigest, otherDigest, msgIsStable);
+      CheckpointMsg *x = new CheckpointMsg(
```

Comment:
x is created using new and then assigned to tableOfStableCheckpoints.
tableOfStableCheckpoints is std::map<ReplicaId, CheckpointMsg*>
Should we use std::map<ReplicaId, std::shared_ptr<CheckpointMsg>>

create 'x' using std::make_shared. 
This will help us not to worry about the memory leak as I am not sure how is the delete called for each of these new.

Reply:
I think you are right. I think we should not to approve any PRs which use raw pointer allocations, unless there is a very good reason for that. Since we are hardly deal with low level programming, I don't see any reason for raw pointer allocations at all.
This one belongs to do original code.
I'll fix it.

Reply:
Specifically here, I need to see how hard it will be to make the change, as there a whole logic around deleting the pointer, and I don't want to get into here. If I can't do it here, I might add a new ticket and assign to one of the managers.

Reply:
@arc-vmware , I was able to convert it to a shared pointer (and removed some code accordingly)